packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -0,0 +1,163 @@
+     * @param value The current value of the input box.
+     * @return Return `undefined`, or the empty string when 'value' is valid.
+     */
+    validateInput?(value: string): string | undefined | PromiseLike<string | undefined>;
+}
+
+@injectable()
+export class MonacoQuickInputService extends MonacoQuickOpenService {
+    private static promptMessage = "Press 'Enter' to confirm your input or 'Escape' to cancel";
+    private options: QuickInputOptions;
+    constructor(@inject(ILogger) logger: ILogger) {
+        super(logger);
+
+    }
+
+    input(options: QuickInputOptions): Promise<string | undefined> {
+        this.options = options;
+        this.options.prompt = this.createPrompt(options.prompt);
+
+        const inputItem = new InputOpenItemOptions(this.options.prompt);
+        this.open({
+            onType: (s, a) => this.validateInput(s, a, inputItem)
+        }, {
+                prefix: options.value,
+                placeholder: options.placeHolder,
+                onClose: () => inputItem.resolve(undefined)
+            });
+        if (options.password) {
+            this.widget.setPassword(true);
+        }
+
+        return new Promise(r => {
+            inputItem.resolve = r;
+        });
+    }
+
+    private createPrompt(prompt?: string): string {
+        if (prompt) {
+            return `${prompt} (${MonacoQuickInputService.promptMessage})`;
+        } else {
+            return MonacoQuickInputService.promptMessage;
+        }
+    }
+
+    private validateInput(str: string, acceptor: (items: QuickOpenItem[]) => void, inputItem: InputOpenItemOptions): void {
+        inputItem.currentText = str;
+        acceptor([new QuickOpenItem(inputItem)]);
+        if (this.options.validateInput) {
+            const hint = this.options.validateInput(str);
+            if (hint) {
+                if (typeof hint !== 'string') {
+                    hint.then(p => {
+                        if (p) {
+                            this.setValidationState(inputItem, p, false);
+                        } else {
+                            this.setValidationState(inputItem, this.options.prompt!, true);
+                        }
+                    });
+                } else {
+                    this.setValidationState(inputItem, hint, false);
+                }
+            } else {
+                this.setValidationState(inputItem, this.options.prompt!, true);
+            }
+        }
+    }
+
+    private setValidationState(inputItem: InputOpenItemOptions, label: string, isValid: boolean): void {
+        this.widget.clearInputDecoration();
+        inputItem.isValid = isValid;
+        inputItem.label = label;
+        this.widget.refresh();
+        if (isValid) {
+            this.widget.clearInputDecoration();
+        } else {
+            this.widget.showInputDecoration(monaco.Severity.Error);
+        }
+    }
+
+    protected get widget(): monaco.quickOpen.QuickOpenWidget {
+        if (this._widget) {
```

Comment:
According to Theia code style we shouldn't use `_` for private fields

Reply:
Simple `widget` name will collide with property getter `widget`. Also see [monaco-quick-open-service.ts](https://github.com/theia-ide/theia/blob/master/packages/monaco/src/browser/monaco-quick-open-service.ts#L56) super class for this class